File: android/src/main/java/np/com/uashim/google_map_polyutil/Transform.java
package np.com.uashim.google_map_polyutil;

import com.google.android.gms.maps.model.LatLng;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class Transform {
    public static List<LatLng> hashMapListToLatLngList(ArrayList<HashMap> hashMapArrayList) {
        List<LatLng> path = new ArrayList<LatLng>();
        for(HashMap p : hashMapArrayList) {
            path.add(hashMapToLatLng(p));
        }
        return path;
    }

    public static LatLng hashMapToLatLng(HashMap hashMap) {
        double lat = (double) hashMap.get("lat");
        double lng = (double) hashMap.get("lng");
      return new LatLng(lat, lng);
    }

    public static List<HashMap> latLngListToHashMap(List<LatLng> latLngList) {
        List<HashMap> hashMaps = new ArrayList<HashMap>();
        for(LatLng point: latLngList) {
            hashMaps.add(latLngToHashMap(point));
        }
        return hashMaps;
    }

    public static HashMap<String, Double> latLngToHashMap(LatLng latLng) {
        HashMap<String, Double> point = new HashMap<>();
        point.put("lat", latLng.latitude);
        point.put("lng", latLng.longitude);
        return point;
    }
}

File: lib/transform.dart
import 'package:google_maps_flutter/google_maps_flutter.dart';

class Transform {
  static Map<String, dynamic> latLngToHashMap(LatLng _latlng) {
    return {"lat": _latlng.latitude, "lng": _latlng.longitude};
  }

  static List<Map<String, dynamic>> listOfLatLngToHashMap(List<LatLng> _list) {
    return _list.map((point) => latLngToHashMap(point)).toList();
  }

  static LatLng hashMapToLatLng(Map<String, dynamic> _map) {
    return LatLng(_map["lat"], _map["lng"]);
  }

  static List<LatLng> listOfHashMapToLatLng(List<Map<String, dynamic>> _list) {
    List<LatLng> res = _list.map((_map) => hashMapToLatLng(_map)).toList();
    return res;
  }
}
